penguin/web/middleware/auth.py

```
@@ -301,3 +308,56 @@ async def protected_endpoint(auth: dict = Depends(require_auth)):
         "subject": request.state.auth_subject,
         "metadata": request.state.auth_metadata
     }
+
+
+def authenticate_connection(
+    connection: Any,
```

Ruff (0.15.10)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `connection`

(ANN401)